src/screens/datasharing/DataSharing.tsx

```diff
@@ -22,10 +22,21 @@ export const DataSharingScreen = () => {
   // if keySubmissionStatus is None we need the 1-time code, otherwise we should go right to consent
   const [isVerified, setIsVerified] = useState(exposureStatus.type === 'diagnosed');
   const [isConfirmedStep1, setIsConfirmedStep1] = useState(false);
-  const onErrorForm = () => {
-    Alert.alert(i18n.translate('DataUpload.FormView.ErrorTitle'), i18n.translate('DataUpload.FormView.ErrorBody'), [
-      {text: i18n.translate('DataUpload.FormView.ErrorAction')},
-    ]);
+  const onErrorForm = (error: any) => {
+    const getTranslationKey = (error: any) => {
+      switch (error.message) {
+        case 'Code 2':
```

Comment:
Can we utilize the types here and just use the error # 
```
enum ErrorCode {
    NONE = 0;
    UNKNOWN = 1;
    INVALID_KEYPAIR = 2;
    DECRYPTION_FAILED = 3;
    INVALID_PAYLOAD = 4;
    SERVER_ERROR = 5;
    INVALID_CRYPTO_PARAMETERS = 6;
    TOO_MANY_KEYS = 7;
    // The timestamp in the Upload message must be no more than one hour old,
    // otherwise this error is generated.
    INVALID_TIMESTAMP = 8;
    INVALID_ROLLING_PERIOD = 10;
    INVALID_KEY_DATA = 11;
    INVALID_ROLLING_START_INTERVAL_NUMBER = 12;
    INVALID_TRANSMISSION_RISK_LEVEL = 13;
    NO_KEYS_IN_PAYLOAD = 14;
  }
```

Reply:
Those are actually the errors for TEK submission response, these are the ones for a diagnosis key claim: https://github.com/cds-snc/covid-alert-app/blob/master/src/services/BackendService/covidshield/covidshield.proto#L27-L36